2024/calcite-react/demo/src/main.tsx

```diff
@@ -16,12 +16,17 @@ import {
   panelHeading,
 } from './config/application.json';
 
-// Step 4: Calcite Components Assets (Recommended to use CDN based on documentation)
-// Please note: You can also manage assets locally by copying assets directly from Calcite Components in your project.
-import { setAssetPath } from '@esri/calcite-components/dist/components/index';
-setAssetPath(
-  'https://cdn.jsdelivr.net/npm/@esri/calcite-components/dist/calcite/assets',
-);
+// Step 4: Calcite Components
+// 4a. Use setAssetPath to the path where assets were copied (occurs in package.json)
+import { setAssetPath } from '@esri/calcite-components/dist/components';
+const { href } = new URL(`${window.location.href}assets`);
+setAssetPath(href);
+
+// 4b. Use defineCustomElements to load calcite components into the app
+import { defineCustomElements } from '@esri/calcite-components/dist/loader';
+defineCustomElements(window, {
```

Comment:
if you use defineCustomElements, then you don't need `setAssetPath` at all, don't you?
and so if you don't need `setAssetPath`, then you don't need to copy the assets?

Reply:
which approach is recommended here?

Reply:
defineCustomElements is simpler, but performance might be a bit worse (won't matter too much in most cases)

in R2 we will improve things by:
- fixing setAssetPath collision
- making calling setAssetPath optional
- remove the need for importing component twice to use it, so then we will start to recommend this

with those fixes, defineCustomElements won't be recommended by us anymore - but until R2, it's the best option